ui/components/app/confirm/info/row/alert-row/alert-row.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { createContext } from 'react';
+import {
+  Severity,
+  TextColor,
+} from '../../../../../../helpers/constants/design-system';
+import InlineAlert from '../../../../confirmations/alerts/inline-alert/inline-alert';
+import useAlerts from '../../../../../../hooks/useAlerts';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowProps,
+  ConfirmInfoRowVariant,
+} from '../row';
+
+export type AlertRowProps = ConfirmInfoRowProps & {
+  alertKey: string;
+  alertOwnerId: string;
+};
+
+export function getAlertTextColors(variant: ConfirmInfoRowVariant): TextColor {
+  switch (variant) {
+    case ConfirmInfoRowVariant.Critical:
+      return TextColor.errorDefault;
+    case ConfirmInfoRowVariant.Warning:
+      return TextColor.warningDefault;
+    case ConfirmInfoRowVariant.Default:
+      return TextColor.infoDefault;
+    default:
+      return TextColor.textDefault;
+  }
+}
+
+function getSeverityAlerts(variant: ConfirmInfoRowVariant): Severity {
+  switch (variant) {
+    case ConfirmInfoRowVariant.Critical:
+      return Severity.Danger;
+    case ConfirmInfoRowVariant.Warning:
+      return Severity.Warning;
+    default:
+      return Severity.Info;
+  }
+}
+
+export const InlineAlertContext = createContext<React.ReactNode | null>(null);
+
+export const AlertRow = ({
+  alertKey,
+  alertOwnerId,
+  children,
+  label,
+  tooltip,
+  variant = ConfirmInfoRowVariant.Default,
+  style,
+}: AlertRowProps) => {
+  const { getFieldAlerts } = useAlerts(alertOwnerId);
+  const hasFieldAlert = getFieldAlerts(alertKey).length > 0;
+
+  const confirmInfoRowProps = {
+    children,
+    label,
```

Comment:
We're still coupled to all the original row properties, meaning we'd have to add them to the list if it ever gets more.

Could we instead change the arguments to `AlertRow` to just `props` then extract only the alert specific properties?

```
const { alertKey, alertOwnerId, ...rowProperties} = props;

...

return <ConfirmInfoRow {...rowProperties} ... />
```

Reply:
Sure, done here https://github.com/MetaMask/metamask-extension/pull/23719/commits/3acec3fe92a4c49aa1fe76163e18eda10c021084